### ISSUE TITLE
Polymorphic elements in logical model

### ISSUE THREAD

**Issue by rlindstrm** [Xt-EHR/xt-ehr-common#9]
How to convey "data type A or data type B" in a logical model?

It looks particularly ugly when using a mixture of FHIR and self-defined data types:
![image](https://github.com/user-attachments/assets/4941f87b-7ba9-4d6c-ba50-f672150278bd)

Options:
1) We use polymorphic data types even when it's ugly
2) We always write out all options with the class allowing only one subelement to be present at the time.

**Comment by rlindstrm** [comment]
Discussed in 13.09 meeting.
Using FHIR data types and logical models the [x] approach looks prettier and could be used. We will try it. We have, for now, moved away from self-declared data types, but the issue should be revisited if we decide to use our own data types.

**Comment by rlindstrm** [comment]
Might be a good idea to capture this decision in the authoring conventions

**Comment by rlindstrm** [comment]
22.10.2024 decision
We leave it as it is for now ([x] approach with models as data types). Hopefully we will find a solution for display. References used only when you can be absolutely sure that this is only a reference (which we might never be able to).

**Comment by rlindstrm** [comment]
@costateixeira 
Reading [this thread](https://chat.fhir.org/#narrow/channel/215610-shorthand/topic/sushi.20handling.20of.20polymorphic.20elements.20in.20data.20models), I am not sure if "type = EHDSModelName" should be in all the models or not? I'm eager to fix, but not sure I understand which way.

**Comment by costateixeira** [comment]
No, the type should not be added. This will be addressed in the tooling and when done we can rever the changes I did.

**Comment by costateixeira** [comment]
https://github.com/HL7/fhir-ig-publisher/issues/978

**Comment by rlindstrm** [comment]
I removed the type when I was cleaning up the qa. I was afraid I was overly eager to do something that had a negative impact instead. Please give us a ping when you notice the update in the Publisher.
